BossMod/Framework/ActionManagerEx.cs

```diff
@@ -167,12 +167,18 @@ public float GCD()
         return gcd->Total - gcd->Elapsed;
     }
 
-    public ActionID GetDutyAction(ushort slot)
+    public (ActionID Action, byte Charges) GetDutyAction(ushort slot)
     {
         var id = ActionManager.GetDutyActionId(slot);
-        return id != 0 ? new(ActionType.Spell, id) : default;
+        if (id == 0)
+            return default;
+
+        var action = new ActionID(ActionType.Spell, id);
+        var dir = EventFramework.Instance()->GetContentDirector();
+
+        return new(action, dir == null ? default : *((byte*)dir + 1484 + slot));
```

Comment:
This really needs to be added to CS, I'll forget about the existence of this field and stuff will break...